### ISSUE TITLE
[tune] Two basic questions on keyword arguments for metrics.

### ISSUE THREAD

**Issue by AvisekNaug** [ray-project/ray#7502]
Starting to go through the docs. In the [tutorial](https://ray.readthedocs.io/en/latest/tune-tutorial.html), it is not apparent how information is passed from the objective function, in this case: train_mnist to the tuner. If the track.log method is passing that information, what arguments does it accept? The says that it can take **kwargs but there should be definite argument names that should exist, shouldn't it? For example, I can pass track.log(redundantmetric=acc, redundandtmetric2=1). How does the search algorithm understand what metric it is trying to optimize in this example. Also, had a similar question with the ray report for function based API. Cannot find any doc on what arguments it takes, which object gets those values and how a search algorithm receives information about a metric when a custom objective function does not return anything. Instea in the body of the objective function, a reporter obejct is called.

In summary, some details on what arguments the tracker.log and the reporter takes and how are they passed to the search algos would be helpful.
